controller/machine/controller.go

```diff
@@ -300,3 +315,30 @@ func (m *Lifecycle) saveConfig(config *machineconfig.MachineConfig, machineDir s
 
 	return obj, nil
 }
+
+func (m *Lifecycle) isNodeInAppliedSpec(machine *v3.Machine) (bool, error) {
+	cluster, err := m.clusterLister.Get("", machine.Spec.ClusterName)
+	if err != nil {
+		return false, err
+	}
+	if cluster == nil {
+		return false, nil
+	}
+	if cluster.DeletionTimestamp != nil {
+		return false, nil
+	}
+	if cluster.Status.AppliedSpec.RancherKubernetesEngineConfig == nil {
+		return false, nil
+	}
+
+	for _, node := range cluster.Status.AppliedSpec.RancherKubernetesEngineConfig.Nodes {
+		machineName := strings.Split(node.MachineName, ":")
```

Comment:
You should check the full string.  The format is `machine.Namespace + ":" + machine.Name`